File: plugin/eval/fixtures/python-patterns/tree/ports.py
"""Port parsing helpers."""

from __future__ import annotations


def parse_port(text: str) -> int:
    """Parse ``text`` as a TCP port in the range 1-65535."""
    try:
        value = int(text.strip())
    except ValueError as exc:
        raise ValueError(f"not a port: {text!r}") from exc
    if not 1 <= value <= 65535:
        raise ValueError(f"port out of range: {value}")
    return value

File: plugin/eval/fixtures/ruby-patterns/tree/lib/ports.rb
# frozen_string_literal: true

# Port parsing helpers.
module Ports
  class Error < StandardError; end

  RANGE = (1..65_535)

  # Parse +text+ as a TCP port in the range 1-65535.
  def self.parse_port(text)
    trimmed = text.to_s.strip
    raise Error, "not a port: #{text.inspect}" unless trimmed.match?(/\A\d+\z/)

    value = Integer(trimmed, 10)
    raise Error, "port out of range: #{value}" unless RANGE.cover?(value)

    value
  end
end
